File: frontend/src/services/admin/warehouse.service.js
import apiAdmin from './apiAdmin.service';
const END_POINT = {
    WAREHOUSES: 'warehouses',
};

export const getAllWareHouses = async () => {
    return await apiAdmin.get(`${END_POINT.WAREHOUSES}`);
};
export const addWareHouse = async (data) => {
    return await apiAdmin.post(`${END_POINT.WAREHOUSES}`, data);
};
export const deleteWareHouse = async (id) => {
    return await apiAdmin.delete(`${END_POINT.WAREHOUSES}/${id}`);
};
export const updateWareHouse = async (data) => {
    return await apiAdmin.put(`${END_POINT.WAREHOUSES}/${data.id}`, data);
};
export const getWareHouseById = async (id) => {
    return await apiAdmin.get(`${END_POINT.WAREHOUSES}/${id}`);
};

File: BackendAPI/Services/WareHouseService.cs
using BackendAPI.Data;
using BackendAPI.Interfaces;
using BackendAPI.UnitOfWorks;

namespace BackendAPI.Services
{
    public class WareHouseService : IWareHouseService
    {
        private readonly IUnitOfWork _unitOfWork;
        public WareHouseService(IUnitOfWork unitOfWork)
        {
            _unitOfWork = unitOfWork;
        }
        public async Task<IEnumerable<WareHouse>> GetAll()
        {
            return await _unitOfWork.GetRepository<WareHouse>().GetAll(orderBy: x => x.OrderByDescending(x => x.Id));
        }
        public async Task<IEnumerable<WareHouse>> GetPagedList(int page, int limit)
        {
            return await _unitOfWork.GetRepository<WareHouse>().GetPagedList(null, null, null, page, limit);
        }
        public async Task<WareHouse?> GetWareHouseById(int id)
        {
            return await _unitOfWork.GetRepository<WareHouse>().GetByID(id);
        }

        public async Task CreateWareHouse(WareHouse newWareHouse)
        {
            await _unitOfWork.GetRepository<WareHouse>().Insert(newWareHouse);
        }
        public async Task UpdateWareHouse(int id, WareHouse updateWareHouse)
        {
            await _unitOfWork.GetRepository<WareHouse>().Update(updateWareHouse);
        }
        public async Task DeleteWareHouse(int id)
        {
            await _unitOfWork.GetRepository<WareHouse>().Delete(id);
        }
    }
}
